### ISSUE TITLE
Cannot import JSON schema

### ISSUE THREAD

**Issue by awdDev786** [Agontuk/schema-designer#2]
I exported JSON schema and then deleted all tables from database.then i tried importing same JSON schema ,it isn't giving any error but not importing schema too.
I tried following solutions before importing schema
     
      

-  deleted previous db and created new with same name and imported

- deleted tables from existing db and imported

- no change in existing db and tried importing
  
  etc
But no solution worked.

Kindly guide me how to import the JSON schema or help me in identifying my mistake.

**Comment by Agontuk** [comment]
Can you open the console and check if it's giving any errors after import? Also check the json file if it contains the tables you created.

**Comment by awdDev786** [comment]
yes there is no error in console and even json file contains the tables i created.

**Comment by awdDev786** [comment]
I tried importing on <a href="https://agontuk.github.io/schema-designer/">Official Demo site</a> too but it didn't work.Kindly share example schema.json file which is working on mentioned link so that i can compare with my schema.json for identifying error.

**Comment by Agontuk** [comment]
Works fine for me, here's a sample json file.

[schema.json.zip](https://github.com/Agontuk/schema-designer/files/1290344/schema.json.zip)

**Comment by awdDev786** [comment]
it is neither showing any schema nor giving error.

**Comment by Agontuk** [comment]
Then probably it's a browser issue, try clearing cache. It's working fine for me. Also if you can, give me a sample json file so that I can test it.

**Comment by awdDev786** [comment]
Google version : 60.0.3112.113.
Windows:10 64bit
database name is same as in schema.json.tried with cache cleaning and the same schema.json file u sent.

**Comment by awdDev786** [comment]
I have sent schema.json at mail.kindly have a look.

**Comment by awdDev786** [comment]
Have you got sample schema which i've sent in email?

**Comment by Agontuk** [comment]
Got the mail, seems like the one I sent. Again it works here. Can you import the json file again and after it is imported, go to console and use this command 
```js
JSON.parse(localStorage.getItem('schema'))
```
The output should match the json file, if not that's the issue you're facing IMO.

**Comment by awdDev786** [comment]
ok thank u for lead.m checking it.

**Comment by awdDev786** [comment]
its empty schema without any table while schema.json has tables.

**Comment by Agontuk** [comment]
Another thing, when you're building a schema and then refresh the page, does it start from the beginning or it persists all the work you did until now ?

**Comment by awdDev786** [comment]
it persists when i refreshed page

**Comment by Agontuk** [comment]
Then it's certainly an importing issue, how are you importing it ? Just clicking import and select the json file, right ? Cause there's no other way as far as I remember.

**Comment by Agontuk** [comment]
Can you try importing files with different extensions and see if it gives an alert like 'Please select the exported json file'

**Comment by awdDev786** [comment]
yes it gave this error once.i changed the db name and kept them same.error vanished nw have no clue

**Comment by Agontuk** [comment]
I've no clue as to what's happening on your side, I tried many times on my side and it works every time. So I haven't been able to reproduce it once. As a last resort, you can try with different browsers and see if it works.

**Comment by EagerDev** [comment]
I am also facing the same issue and one more observation is i am passing a value from header to schema component where it is updating the localstorage.but change in localstorage isn't updating component.I think both issues are same that change in localstorage after component is already rendered isnt re-rendering the component.

**Comment by Agontuk** [comment]
localStorage change doesn't trigger re-rendering. It's only purpose is to read the saved state when you reload the page.

**Comment by awdDev786** [comment]
@Agontuk  kindly look at this video of issue.I made it for you.
<a href="https://www.youtube.com/watch?v=ZpxEm9mJI_I&feature=youtu.be">ISSUE</a>  is described here.

My browser is:
Google Chrome | 61.0.3163.100

**Comment by Agontuk** [comment]
Thanks for the video, it looks like your browser is not reloading itself after importing the json file. I'll take a look at it ASAP.

**Comment by Agontuk** [comment]
I pushed a possible fix for this issue, let me know if it works for you.

**Comment by awdDev786** [comment]
Have you updated in official site <a href="https://agontuk.github.io/schema-designer/">here</a>?

**Comment by awdDev786** [comment]
Official Demo is still having the same issue.and also cloning new version didn't do the trick.(Sorry for accidentally closing it.)

**Comment by Agontuk** [comment]
@awdDev786 can you check if [this](https://codepen.io/matt-west/pen/KjEHg) snippet works in your browser? I think FileReader API is somehow not triggering any event on your end.

**Comment by awdDev786** [comment]
Through this code only .txt files can be read.schema.json file is not being read.

**Comment by Agontuk** [comment]
So, it worked with .txt files ?

**Comment by awdDev786** [comment]
yes.it read .txt file

**Comment by Agontuk** [comment]
Thanks, I'll try another fix.

**Comment by awdDev786** [comment]
Ok thank you for your concern.

**Comment by Agontuk** [comment]
@awdDev786 can you clone this and try adding a console.log statement [here](https://github.com/Agontuk/schema-designer/blob/master/src/components/ImportDatabase.js#L11). Then see if it logs to console when you import a file. I thought FileReader API is the culprit, but now I think file input change is not triggering on your end.

**Comment by awdDev786** [comment]
@Agontuk  not working.same behavior.

**Comment by Agontuk** [comment]
So, `handleFileUpload` method is not getting called ?

**Comment by awdDev786** [comment]
![image](https://user-images.githubusercontent.com/29519534/32055006-d806bef4-ba79-11e7-92ea-26d9b3f7fb58.png)
  

File is being uploaded and read here.

**Comment by awdDev786** [comment]
sorry for bothering you again,can u find some solution of this issue?

**Comment by Agontuk** [comment]
I've pushed another possible fix, can you try it locally by cloning the repo? If it doesn't work, try putting a console.log [here](https://github.com/Agontuk/schema-designer/blob/master/src/components/ImportDatabase.js#L20) and [here](https://github.com/Agontuk/schema-designer/blob/master/src/components/ImportDatabase.js#L24). See if they get logged in the console.

**Comment by awdDev786** [comment]
@Agontuk  empty console in both locations.

**Comment by Agontuk** [comment]
@awdDev786 I'm out of ideas about why it's not working in your system. I'll try to investigate it more and see if I can reproduce it in my system. Please let me know if you have any ideas.

**Comment by awdDev786** [comment]
@Agontuk Kindly update your online demo so that we can try there too.

**Comment by Agontuk** [comment]
Done, please let me know if you figure out what's the issue.

**Comment by awdDev786** [comment]
- I thought it can be an issue with my local deployment but same isn't working in official demo which means not an issue with deployment.

- I also deployed it on a vps same issue means not an issue with my environment.

- Additionally also tried with ms edge browser and mozilla so no issue with chrome.

- Possibility is may be an issue with file import export.
    I made a schema exported it and deleted schema from designer and checked the schema.json file,Correct schema is exported.when i imported it in the same designer it firstly gave error as use exported schema and then it gave no error also empty schema in the response of   
     JSON.parse(window.localStorage.getItem('schema'))

- Possiblity of fileupload is also checked before that no file is being uploaded.

So @Agontuk Kindly suggest me an other work around, if there is any other possibility i can check.

Thankx for your time.

**Comment by Agontuk** [comment]
What do you mean by "it firstly gave error"? That should never happen, unless you import files which don't have `.json` extension. Since the file you exported is a json file, it shouldn't occur. Can you explain this scenario in more detail?

**Comment by awdDev786** [comment]
Yes file is with .json form but as we have already checked by console that file is not being read here so it is not getting any file that's y it is giving such error.

**Comment by Agontuk** [comment]
I'm checking the file type before reading it, so it shouldn't be giving this error. Anyway I just pushed a fix for an issue you mentioned on fourth point. Can you check locally if it changes anything on your end?

**Comment by awdDev786** [comment]
No it has same issues.

**Comment by Agontuk** [comment]
From your previous screenshot, I'm seeing `file.type` is empty. That's why it's giving the error. I'll check why it's being empty and make a fix asap.

**Comment by Agontuk** [comment]
I'm thinking of exporting the file as a .txt instead of .json, will let you know when it's done.

**Comment by Agontuk** [comment]
@awdDev786, I've changed the import & export feature to use .txt file. Can you rename your schema.json to schema.txt and see if it works locally?

**Comment by awdDev786** [comment]
@Agontuk issue was resolved as your updated code. You have used this code to display imported schema in display.
                          window.location.reload();
I tried replacing it with some solution as i only want to reload/refresh <Tables > component from schema.js despite of full window reload.Is it possible as i think its isn't good practice to reload window on schema changes.
